### PR TITLE
chore(datasets): Normalise optional requirements names and move them to `pyproject.toml`

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description

Proposal for #313 

## Development notes

- Nomalised all optional requirements in accordance with PEP 685
- Move all requirements to `pyproject.toml`
- Remove `setup.py`

If we're happy with the naming already - we can merge this and release `kedro-datasets 3.0.0`
Follow up actions required -
* Release `kedro-datasets`
* Update docs on Kedro
* Update `requirements.txt` in `kedro-starters`
* Unpin pip in most places that we've pinned it in
## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-datasets/pyproject.toml [hunk 1]
```diff
@@ -14,7 +14,223 @@ dependencies = [
...
+dask-parquetdataset = ["dask[complete]>=2021.10", "triad>=0.6.7, <1.0"]
+dask = ["kedro-datasets[dask-parquetdataset]"]
+
+databricks-managedtabledataset = ["kedro-datasets[spark-base,pandas-base,delta-base]"]
+databricks = ["kedro-datasets[databricks-managedtabledataset]"]
+
+geopandas-geojasondataset = ["geopandas>=0.6.0, <1.0", "pyproj~=3.0"]
+geopandas = ["kedro-datasets[geopandas-geopandasjsondataset]"]
+
+holoviews-holovieswriter = ["holoviews~=1.13.0"]
```
Comment (ankatiyar) [inline kedro-datasets/pyproject.toml:45]:
```suggestion
holoviews-holoviewswriter = ["holoviews~=1.13.0"]
```

**Review by astrojuanlu — APPROVED** [review]
I'm +1 on basically turning all the extras groups to lowercase and replace `.` by `-` 💯 

We'd need to update our docs, starters etc but at least I agree with this solution

**Comment by lrcouto** [comment]
> I'm +1 on basically turning all the extras groups to lowercase and replace `.` by `-` 💯
> 
> We'd need to update our docs, starters etc but at least I agree with this solution

I agree. It's a pretty straightforward solution. It'd need the aforementioned housecleaning on docs, starters, etc. but I like it.

**Review by lrcouto — APPROVED** [review]

**Comment by astrojuanlu** [comment]
We decided to hold this until kedro-datasets 2.1.0 is released 👍🏽

**Review by SajidAlamQB — APPROVED** [review]

**Comment by ankatiyar** [comment]
Reminder to do another pass at `setup.py` to make sure all versions are up to date - https://github.com/kedro-org/kedro-plugins/pull/537

**Review by noklam — APPROVED** [review]

**Pull request merged**